text/entities/0264-resource-and-entities.md

```diff
@@ -0,0 +1,852 @@
+
+The protocol today requires a raw grab bag of Attributes on Resource. We cannot break this going forward.  However, Entities represent a new mechanism of "bundling" attributes on Resource and interacting with these bundles.  We do not want this to bloat the protocol, nor do we want it to cause oddities.
+
+Going forward, we have set of options:
+
+- Duplicate attributes in `Entity` section of Resource.
+- Reference attributes of Resource in entity.
+- Only identify Entity id and keep attribute<->entity association out of band.
+- Extend Attribute on Resource so that we can track the entity type per Key-Value (across any attribute in OTLP).
+
+The third option prevents generic code from interacting with Resource and Entity without understanding the model of each.  The first keeps all usage of entity simple at the expense of duplicating information and the middle is awkward to interact with from an OTLP usage perspective. The fourth is violates our stability policy for OTLP.
+
+### How to deal with Resource/Entities whose lifecycle does not match the SDK?
+
+This proposal motivates a Resource Provider in the SDK whose job could include managing changes in entity lifetimes, but does not account for how these changes would be broadcast across TracerProvider, LogProvider, MeterProvider, etc.  That would be addressed in a follow on OTEP.
+
+### How to deal with Prometheus Compatibility for non-SDK telemetry?
+
+Today, [Prometheus compatibility](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/compatibility/prometheus_and_openmetrics.md) relies on two key attributes in Resource: `service.name` and `service.instance.id`. These are not guaranteed to exist outside of OpenTelemetry SDK generation. While this question is not fully answered, we believe outlining identity in all resources within OpenTelemetry allows us to define a solution in the future while preserving compatibility with what works today.
+
+Here's a list of requirements for the solution:
+
+- Existing prometheus/OpenTelemetry users should be able to migrate from where they are today.
+- Prometheus should have a set of identifying attributes for their up and coming `info()` function and info metric features.
+- (desired) OpenTelemetry should be able to create unique `job`/`instance` labels between all metrics sent to prometheus for any "info" metric join.
+
+A quick proposal of what this might look like:
+
+- `target_info` metric generation is updated to exclude any keys which are contained in `descriptive_attributes_keys` of an entity.
+- For each entity which has non-empty descriptive_attributes_keys, generate an info metric: `<entity_type>_entity_info` (naming TBD), which has all identifying and descriptive keys. This should play nicely with the planned improvements to [info-typed metrics](https://github.com/prometheus/proposals/blob/main/proposals/2024-04-10-native-support-for-info-metrics-metadata.md#goals).
```

Comment:
> For each entity which has non-empty descriptive_attributes_keys, generate an info metric: `<entity_type>_entity_info` (naming TBD)

Isn't this really the same as the option in the paragraph below ("that would break compatibility")? I wonder if this paragraph instead should mention synthesizing a `target_info` `instance` label from all the identifying resource attributes.